cli/src/config.rs

```diff
@@ -584,8 +584,8 @@ mod tests {
             command_args,
             CommandNameAndArgs::Structured {
                 env: hashmap! {
-                    "KEY1".to_string() => "value1".to_string(),
-                    "KEY2".to_string() => "value2".to_string(),
+                    "key1".to_string() => "value1".to_string(),
+                    "key2".to_string() => "value2".to_string(),
```

Comment:
> environment config variables are no longer case-sensitive

Oops, it probably means the structured command config no longer works. Environment variables are case sensitive on Unix, and we do use the TOML table syntax to override them.

Maybe it's time to roll our own config module? I think it will help implement #616 and/or handle colors/revset-aliases overrides better.

Sorry for the late reply. I didn't notice this yesterday.

Reply:
Hmm, I think my summary of what changed in `config` v0.14.0 is not quite right.  Prior to v0.14.0 environment variable keys [were already being downcased](https://github.com/mehcode/config-rs/blob/v0.13.4/src/env.rs#L181), see https://github.com/mehcode/config-rs/issues/340, but the internal keys tracked by `config` were not. For example, an internal `PAGER` key would not be set by environment variable `PAGER` because the latter had been converted to `pager`.

The change in https://github.com/mehcode/config-rs/pull/354 was to downcase the internal keys as well. Looking at the config schema I don't see any mixed case settings, do you have an example of what you would expect to break so I can be sure I understand?

Reply:
> Prior to v0.14.0 environment variable keys [were already being downcased](https://github.com/mehcode/config-rs/blob/v0.13.4/src/env.rs#L181)

I think you're talking about the config loaded from environment variables, but this `CommandNameAndArgs` stuff is data configured in a config file, and passed to the external process. So the case matters.

> do you have an example of what you would expect to break

In addition to this, `[aliases]`, `[revset-aliases]`, `[template-aliases]`, `[merge-tools.<NAME>]` would break if keys are downcased. They are free-form dictionary, and users may define `revset-aliases.HEAD = "@-"` for example.

Reply:
@wfchandler: To clarify, I think Yuya is talking about https://github.com/martinvonz/jj/blob/26528091e646a79a3724997f06c4af69eaf70297/cli/src/config/misc.toml#L9 (the user can have their own similar config). (Correct me if I'm wrong, @yuja.)

Reply:
Perhaps, this is the upstream issue. It doesn't describe the exact problem we have, but the root cause would be the same.
https://github.com/mehcode/config-rs/issues/531

Reply:
Thanks for the explanation, I see what you mean.  I'll go ahead and close out this PR given the problem with `config` v0.14.0.

Thank you @yuja @martinvonz and @jonathantanmy for the thorough review, much better to have caught this issue now.

Reply:
I hope the problem will be addressed at config-rs, and this PR can be revisited later. Thanks anyways.

Reply:
Yes, thanks for trying, and sorry about the trouble. And thanks to Yuya for noticing the problem!